In [26]:
import pandas as pd
df = pd.read_csv('sentiment_scores.csv')

In [27]:
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report, cohen_kappa_score

# These are the true sentiment scores calculate based on voteed_up, votes_up, votes_funny and weighted_vote_score
true_scores = df['true_scores']

# These are the sentiment scores predicted by your BERT model
predicted_scores = df['sentiment_score']

# Calculate the confusion matrix
conf_matrix = confusion_matrix(true_scores, predicted_scores)

# Calculate the accuracy of the predictions
accuracy = accuracy_score(true_scores, predicted_scores)

# Generate a classification report
class_report = classification_report(true_scores, predicted_scores)

# Calculate Cohen's Kappa
kappa_score = cohen_kappa_score(true_scores, predicted_scores)

print("Confusion Matrix:\n", conf_matrix)
print("\nAccuracy:", accuracy)
print("\nClassification Report:\n", class_report)
print("\nCohen's Kappa:", kappa_score)


Confusion Matrix:
 [[31143     0     0     0  5564]
 [ 4248 17193     0     0     0]
 [    0  2266 21701     0     0]
 [    0     0  2988 31139     0]
 [    0     0     0  4293 40788]]

Accuracy: 0.8799985123014077

Classification Report:
               precision    recall  f1-score   support

           1       0.88      0.85      0.86     36707
           2       0.88      0.80      0.84     21441
           3       0.88      0.91      0.89     23967
           4       0.88      0.91      0.90     34127
           5       0.88      0.90      0.89     45081

    accuracy                           0.88    161323
   macro avg       0.88      0.87      0.88    161323
weighted avg       0.88      0.88      0.88    161323


Cohen's Kappa: 0.8470463425025325


High Accuracy: The bert model achieved an accuracy of approximately 88%, which means it correctly predicted the sentiment scores for 88% of the reviews in your dataset.

Balanced Performance Across Categories: The precision, recall, and F1-scores are consistently high across all sentiment categories (1 to 5), suggesting that the model is reliably identifying the sentiment of the reviews across the spectrum, without significant bias toward any specific sentiment score.

Confusion Matrix Insights: Most errors seem to occur between adjacent sentiment categories, which is common in sentiment analysis due to the subjective nature of sentiment.

Strong Agreement (Cohen's Kappa): The Cohen's Kappa score of 0.847 indicates a strong agreement between the model’s predictions and the actual sentiment scores, suggesting that the model's assessments are consistent with the expected outcomes.